### ISSUE TITLE
bundle init undocumented

### ISSUE THREAD

**Issue by burningTyger** [bundler/bundler#6140]
I ran bundle init becaue I always forget what to put in the first line.
Some documentation would be nice.

I'm confronted with this:

````ruby
# frozen_string_literal: true
# => Do I need this line?

source "https://rubygems.org"
# => I know what this is.

git_source(:github) {|repo_name| "https://github.com/#{repo_name}" }
# => What's this good for? Never seen before

# gem "rails"
# => ok, know what this is too

**Comment by segiddins** [comment]
See http://bundler.io/v1.16/man/bundle-init.1.html for documentation for the init command

**Comment by burningTyger** [comment]
The page doesn't really tell what `bundle init` puts into the Gemfile. I'm curious, I checked that page before I posted this issue.

**Comment by segiddins** [comment]
Sorry, I had just read the title of your issue

**Comment by arbonap** [comment]
@burningTyger there's Gemfile documentation that explains parts of the Gemfile much more in depth. [Here's documentation](http://bundler.io/v1.16/man/gemfile.5.html#GIT-SOURCE) that explains how to add a custom Git Source. Maybe adding a link to Gemfile documentation in the `bundle-init` doc would be helpful.

The comment `# frozen_string_literal: true` tells Ruby that all string literals in the file are implicitly frozen. This is in preparation for Ruby 3, where all string literals are going to be the default. I'd be happy to add some documentation explaining this.

**Comment by segiddins** [comment]
@arbonap that'd be cool!

**Comment by burningTyger** [comment]
@arbonap thank you for pointing this out. Makes sense now. Adding this would be great.

**Comment by rubymorillo** [comment]
@arbonap hey Patricia, let me know if you need help adding this to the `bundle-init` man page!

**Comment by nathandibley** [comment]
I'd like to pick this one up.
I'll use these docs as a guide - http://bundler.io/doc/documentation/writing.html

To confirm, the desire is to have a link to http://bundler.io/man/gemfile.5.html#GIT-SOURCE, and also some short explanation on why `# frozen_string_literal: true` is in the default Gemfile added to the `bundle-init` man page ?

**Comment by 315tky** [comment]
I'd like to pick this one up.
I'll use these docs as a guide - http://bundler.io/doc/documentation/writing.html

To confirm, the desire is to have a link to http://bundler.io/man/gemfile.5.html#GIT-SOURCE, and also some short explanation on why # frozen_string_literal: true is in the default Gemfile added to the `bundle-init` man page ?

**Comment by 315tky** [comment]
#6302

**Comment by 315tky** [comment]
Just noticed this was still open, let me know if you require any further changes to it

**Comment by indirect** [comment]
Merged in #6302.

**Comment by burningTyger** [comment]
Thanks everybody! Good documentation is worth gold.
